app/jobs/remove_old_throttles_job.rb

```diff
@@ -0,0 +1,43 @@
+class RemoveOldThrottlesJob < ApplicationJob
+  queue_as :low
+
+  WINDOW = 30.days.freeze
+
+  include GoodJob::ActiveJobExtensions::Concurrency
+
+  good_job_control_concurrency_with(
+    enqueue_limit: 1,
+    perform_limit: 1,
+    key: -> do
+      rounded = TimeService.round_time(time: arguments.first, interval: 1.hour)
+      "remove-old-throttles-#{rounded.to_i}"
+    end,
+  )
+
+  discard_on GoodJob::ActiveJobExtensions::Concurrency::ConcurrencyExceededError
+
+  def perform(now, limit: 1000, total_limit: 100_000)
+    max_window = Throttle::THROTTLE_CONFIG.map { |_, config| config[:attempt_window] }.max
+    total_removed = 0
+
+    loop do
+      removed_count = Throttle.
+        where('updated_at < ?', now - (WINDOW + max_window.minutes)).
+        limit(limit).
+        delete_all
+
+      total_removed += removed_count
+
+      Rails.logger.info(
+        {
+          name: 'remove_old_throttles',
+          removed_count: removed_count,
+          total_removed: total_removed,
+        }.to_json,
+      )
```

Comment:
Do we want this to happen inside the loop, or after? Noting since at this point, `total_removed` is more of a running tally.

Reply:
I thought of it like a running tally, could also log the limit so we could see progress against that in the logs

Reply:
added `total_limit` to payload in 026178f9783d69478bb2bcf29be5eb74a63cc2bd